### PR TITLE
Mapper2

### PR DESCRIPTION
**Pull request by rdaly525**
This adds in a mapper test and fixes bitwise errors in the cond.py

### CODE REVIEW

**Comment by rdaly525** [comment]
@leonardt and @alexcarsello, could you review this?

**Comment by rdaly525** [comment]
@rsetaluri, I slightly changed the api to get the instance name -> instruction dict. Check test_mapper.py

File: tests/test_mapper.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            inst.regf == type(inst.regf).BYPASS and
+            (inst.cond == type(inst.cond).Z or inst.cond == type(inst.cond).Z_n)
+        )
+    mapper.add_discover_constraint(bypass_mode)
+    
+    mapper.discover_peak_rewrite_rules(width=16)
+    #test the mapper on simple add4 app
+    app = c.load_from_file("tests/add4.json")
+    mapper.map_app(app)
+    imap = mapper.extract_instr_map(app)
```
Comment (rsetaluri) [inline tests/test_mapper.py]: Left some comments in the latest metamapper PR. I'm confused about this API.

**Comment by leonardt** [comment]
I merged the `bfloat` branch into master then updated this branch and it broke the build. I'm guessing it's because the `bfloat` branch uncommented the floating point code. How should we handle this integration? Wait until float is supported? Or comment the code and maybe create a separate test path for just RTL until the mapper supports it.

**Comment by rdaly525** [comment]
The break was due to a bug (that was fixed) in the MetaMapper.

**Pull request merged**